python/triton/compiler/compiler.py

```diff
@@ -56,7 +56,7 @@ def hash(self):
     "ptx": ptx_prototype_pattern,
 }
 
-mlir_arg_type_pattern = r'%\w+: ((?:[^,\s<]+|<[^>]+>)+),?'
+mlir_arg_type_pattern = r'%\w+: ((?:[^,\s<)]+|<[^>]+>)+),?'
```

Comment:
is this related? (I won't claim I decoded it..)

Reply:
A previous version of this patch needed us to fix the regex for parsing an MLIR
file to find the individual arguments. Previously it would include the
function's closing paren in the argument type. We keep this because it's a
reasonable fix.